### PR TITLE
[Storage] Change first ServiceVersion value to 1 and throw for everything else

### PR DESCRIPTION
**Pull request by ShivangiReja**
[As per API review](https://apiview.azurewebsites.net/Assemblies/Review/c52f059a-6d57-4d40-b7d2-a5816565a660#Azure.Storage.Blobs.BlobClientOptions.ServiceVersion) Change  ServiceVersion value to  `1` and throw exception for everything else.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Common/src/Errors.cs [hunk 1]
```diff
@@ -60,6 +60,9 @@ public static ArgumentException InvalidPermission(char s)
         public static ArgumentException InvalidResourceType(char s)
             => new ArgumentException($"Invalid resource type: '{s}'");
 
+        public static ArgumentException ArgumentNotSupported(string paramName)
+           => new ArgumentException("Argument not supported", paramName);
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Common/src/Errors.cs]: we need a more specific message here like `The service version is not supported`

File: sdk/storage/Azure.Storage.Blobs/src/BlobClientOptions.cs [hunk 2]
```diff
@@ -62,7 +62,7 @@ public enum ServiceVersion
             ServiceVersion version = LatestVersion,
             CustomerProvidedKey? customerProvidedKey = default)
         {
-            Version = version;
+            Version = (int)version == 1? version: throw Errors.ArgumentNotSupported(nameof(version));
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Blobs/src/BlobClientOptions.cs]: Compare with `V2018_11_09` instead of `1`

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/BlobClientOptions.cs [hunk 3]
```diff
@@ -31,7 +31,7 @@ public enum ServiceVersion
             /// The 2018-11-09 service version described at
             /// <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/version-2018-11-09" />
             /// </summary>
-            V2018_11_09 = 0
+            V2018_11_09 = 1
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobClientOptions.cs]: While we're in here making changes, can we please change all of these from `V2018_11_09` to `V2019_02_02`?  We've decided we're going to use that version for Blobs/Files/Queues across all our languages.  Please update the comments and link to https://docs.microsoft.com/en-us/rest/api/storageservices/versioning-for-the-azure-storage-services#version-2019-02-02 as well.

**Review by tg-msft — APPROVED** [review]

**Review by tg-msft — APPROVED** [review]
Awesome - thanks!

**Pull request merged**